store-implementation/hbase-store/src/main/java/uk/gov/gchq/gaffer/hbasestore/SingleUseHBaseStore.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2016-2017 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.hbasestore;
+
+import uk.gov.gchq.gaffer.hbasestore.utils.TableUtils;
+import uk.gov.gchq.gaffer.store.StoreException;
+import uk.gov.gchq.gaffer.store.StoreProperties;
+import uk.gov.gchq.gaffer.store.schema.Schema;
+
+/**
+ * An {@link HBaseStore} that deletes the underlying table each time it is initialised.
+ * Meant to be used for testing.
+ */
+public class SingleUseHBaseStore extends HBaseStore {
```

Comment:
It was added to mirror the SingleUseAccumuloStore. I suppose it is possible that a user wants the table to be deleted every time a new Graph is initialised, seems unusual but lets leave it here for the time being.